### ISSUE TITLE
Issue about validation of pretraining

### ISSUE THREAD

**Issue by mn7ark** [s3prl/s3prl#165]
Hi, I'm currently running the pretraining code. But I found that there is not validation data used to choose the best checkpoint. So should I  just use the newest checkpoint for downstream task?

**Comment by andi611** [comment]
Yes, we did not implement a validation loss to choose the best checkpoint, we simply take the latest, after the loss has converged.

However, I think the validation loss was implemented in this issue #164.
Perhaps a pull request can be made to add this feature!

**Comment by mn7ark** [comment]
Thanks for the reply! 
  By the way, I find that in pretrain/tera/exp_model_config the noise proportion is set
to 0.15, but in tera/config_model.yaml it is set to 0. 
  As the README said, to run the pretraining, config_model.yaml should be used. Is this a bug? and the paper use P_N to denote noise proportion, but I didn't find exact value. Is 0.15 the value for all tera variant with magnitude alteration in the paper?

**Comment by andi611** [comment]
Hi @mn7ark,

> Thanks for the reply!
> By the way, I find that in pretrain/tera/exp_model_config the noise proportion is set
> to 0.15, but in tera/config_model.yaml it is set to 0.
> As the README said, to run the pretraining, config_model.yaml should be used. Is this a bug? 

You can use either, you can pre-train TERA with magnitude alteration (set noise proportion > 0) or without (set noise proportion == 0). The configs under `pretrain/tera/exp_model_config` are the ones I've used to conduct experiments, and the `pretrain/tera/config_model.yaml` is the one I set as default for general purpose.

> and the paper use P_N to denote noise proportion, but I didn't find exact value. Is 0.15 the value for all tera variant with magnitude alteration in the paper?

Yes, it is 0.15, sorry for missing the exact value. All TERA variant with magnitude alteration used this value.

**Comment by mn7ark** [comment]
Thanks, get it! Sorry for the late reply.
